### PR TITLE
Add support for multiple os images per multi-arch tag

### PR DESCRIPTION
**Pull request by MichaelSimons**
These changes allow for manifest tag to reference to more than one image per platform which will be for different OS versions.

This functionality is need as part of https://github.com/dotnet/dotnet-docker/issues/298.

### CODE REVIEW

File: src/Microsoft.DotNet.ImageBuilder/Commands/BuildCommand.cs [hunk 1]
```diff
@@ -35,22 +37,34 @@ private void BuildImages()
...
-                    ExecuteHelper.Execute(
-                        "docker",
-                        $"build {tagArgs} -f {dockerfilePath} {image.ActivePlatform.BuildContextPath}",
-                        Options.IsDryRun);
-                }
-                finally
-                {
-                    if (createdPrivateDockerfile)
+                    string dockerfilePath;
+                    bool createdPrivateDockerfile = UpdateDockerfileFromCommands(platform, out dockerfilePath);
```
Comment (ravimeda) [inline src/Microsoft.DotNet.ImageBuilder/Commands/BuildCommand.cs]: Inline the declaration - `out string dockerfilePath`

Comment (MichaelSimons) [reply src/Microsoft.DotNet.ImageBuilder/Commands/BuildCommand.cs]: Fixed

File: src/Microsoft.DotNet.ImageBuilder/ViewModel/ManifestFilter.cs [hunk 2]
```diff
@@ -20,11 +21,11 @@ public ManifestFilter()
...
         }
 
-        public Platform GetActivePlatform(Image image)
+        public IEnumerable<Platform> GetActivePlatforms(Image image)
         {
             return GetPlatforms(image)
-                .Where(platform => platform.OS == DockerOS && platform.Architecture == DockerArchitecture)
-                .SingleOrDefault();
+                .Where(platform => string.Equals(platform.OS, DockerOS, StringComparison.OrdinalIgnoreCase)
+                    && platform.Architecture == DockerArchitecture);
```
Comment (ravimeda) [inline src/Microsoft.DotNet.ImageBuilder/ViewModel/ManifestFilter.cs]: Ignore case on `platform.Architecture == DockerArchitecture`?

Comment (MichaelSimons) [reply src/Microsoft.DotNet.ImageBuilder/ViewModel/ManifestFilter.cs]: Architecture is an enum 😄

Comment (ravimeda) [reply src/Microsoft.DotNet.ImageBuilder/ViewModel/ManifestFilter.cs]: I need some coffee I guess :)

**Pull request merged**